example/aws/kustomization.yaml

```diff
@@ -8,23 +8,36 @@ patchesStrategicMerge:
   - thanos-query.yaml
   - thanos-rule.yaml
 secretGenerator:
+  # description: config + credentials for access to object storage
+  # used-by: Prometheus, Thanos Store, Thanos Compact
   - name: thanos-storage
     type: Opaque
     files:
       - config.yaml=secrets/thanos-storage-secret.yaml
 configMapGenerator:
+  # description: application alerts handled by thanos-rule
+  # used-by: Thanos Rule
   - name: alerts
     files:
       - prometheus-alerts.yaml=resources/prometheus-alerts.yaml
+  # description: Prometheus configuration
+  # used-by: Prometheus
   - name: prometheus
     files:
       - prometheus.yaml.tmpl=resources/prometheus.yaml.tmpl
+  # description: Alerts to monitor Thanos infrastructure
+  # used-by: Prometheus
   - name: thanos-rule-alerts
     files:
       - thanos-rule.yaml=resources/thanos-rule-alerts.yaml
+  # description: Discovery config for TQ instances used to evaluate alert rules
+  # used-by: Thanos Rule
   - name: thanos
```

Comment:
Comments are great and useful. Looking at general naming, shouldn't this be called `thanos-rule` to mimic `thanos-query` style?

Reply:
Yeah, that would make sense.

I have kept the names backward compatible, but so much else has changed, we might as well break that compatibility.